src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const array = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    transform(stateCopy, action);
+  }
+
+  function transform(result, action) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(result, action.extraData);
+        array.push({ ...stateCopy });
```

Comment:
Pushing the state copy to the array should be done after processing each action, outside of the `switch` block. This ensures that the state is captured after all modifications for the current action are complete.